core/services/vrf/delegate.go

```diff
@@ -1,31 +1,377 @@
 package vrf
 
 import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager"
+
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/gracefulpanic"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/solidity_vrf_coordinator_interface"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
 	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/log"
 	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	"github.com/smartcontractkit/chainlink/core/services/signatures/secp256k1"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"gorm.io/gorm"
 )
 
 type Delegate struct {
-	vorm ORM
-	pr   pipeline.Runner
-	porm pipeline.ORM
+	cfg    Config
+	vorm   ORM
+	db     *gorm.DB
+	pr     pipeline.Runner
+	porm   pipeline.ORM
+	vrfks  *VRFKeyStore
+	gethks GethKeyStore
+	ec     eth.Client
+	lb     log.Broadcaster
+}
+
+//go:generate mockery --name GethKeyStore --output mocks/ --case=underscore
+
+type GethKeyStore interface {
+	GetRoundRobinAddress(addresses ...common.Address) (common.Address, error)
 }
 
-func NewDelegate(vorm ORM, pr pipeline.Runner, porm pipeline.ORM) *Delegate {
+type Config struct {
```

Comment:
The problem with using a struct is that it cuts off runtime config changes if we ever want to do that.

The pattern we have used elsewhere is a `Config` interface that just defines the methods, then you can query it every time you use it.

Reply:
yeah I was just thinking none of these currently support runtime changes and fmv2 also uses a struct like this - happy to just change both though